src/devices/src/virtio/net/persist.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring net devices.
+
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+
+use rate_limiter::{persist::RateLimiterState, RateLimiter};
+use snapshot::Persist;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use super::*;
+
+use crate::virtio::persist::VirtioDeviceState;
+use crate::virtio::{DeviceState, Queue};
+
+#[derive(Versionize)]
+pub struct NetConfigSpace {
+    guest_mac: Vec<u8>,
+}
+
+#[derive(Versionize)]
+pub struct NetState {
+    id: String,
+    tap_if_name: String,
+    rx_rate_limiter_state: RateLimiterState,
+    tx_rate_limiter_state: RateLimiterState,
+    allow_mmds_requests: bool,
+    config_space: NetConfigSpace,
+    virtio_state: VirtioDeviceState,
+}
+
+pub struct NetConstructorArgs {
+    mem: GuestMemoryMmap,
+}
```

Comment:
nit: all devices will also need to take a reference to `EventManager` to register their handlers on `restore()`.

But we can add that incrementally when we get there.